tests/Api/Shop/OrdersTest.php

```diff
@@ -371,4 +371,24 @@ public function it_removes_item_from_the_cart(): void
 
         $this->assertResponseCode($this->client->getResponse(), Response::HTTP_NO_CONTENT);
     }
+
+    /** @test */
+    public function it_returns_unprocessable_entity_status_if_tries_to_remove_an_item_that_not_exist_in_the_order(): void
+    {
+        $this->loadFixturesFromFiles(['channel.yaml', 'cart.yaml']);
+
+        $tokenValue = 'nAWw2jewpA';
+
+        /** @var MessageBusInterface $commandBus */
+        $commandBus = self::getContainer()->get('sylius.command_bus');
+
+        $pickupCartCommand = new PickupCart($tokenValue);
+        $pickupCartCommand->setChannelCode('WEB');
+        $commandBus->dispatch($pickupCartCommand);
+        $nonExistingOrderItemId = 123;
+
+        $this->client->request('DELETE', sprintf('/api/v2/shop/orders/%s/items/%s', $tokenValue, $nonExistingOrderItemId));
+
+        $this->assertResponseCode($this->client->getResponse(), Response::HTTP_UNPROCESSABLE_ENTITY);
```

Comment:
This is a case that the order item resource cannot be found. Shouldn't it be 404 instead?

Reply:
You may be right. I might have overthink it too much, such as the order resource part is correct and its sub-resource is incorrect, hence the status code 422.

Alternatively:
https://www.rfc-editor.org/rfc/rfc4918#section-11.2
"but was unable to process the contained instructions"
Removing something that doesn't exist is unprocessable 💃 

I am not able to clearly state which one is correct 🤔